File: app/src/main/java/com/popalay/yocard/ui/holders/HoldersFragment.java
package com.popalay.yocard.ui.holders;

import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.arellomobile.mvp.presenter.InjectPresenter;
import com.popalay.yocard.R;
import com.popalay.yocard.data.models.Holder;
import com.popalay.yocard.databinding.FragmentHoldersBinding;
import com.popalay.yocard.ui.base.BaseFragment;
import com.popalay.yocard.ui.base.ItemClickListener;
import com.popalay.yocard.ui.holderdetails.HolderDetailsActivity;

import java.util.List;

public class HoldersFragment extends BaseFragment implements HoldersView, ItemClickListener<Holder> {

    @InjectPresenter HoldersPresenter presenter;

    private FragmentHoldersBinding b;

    private HoldersViewModel viewModel;

    public static HoldersFragment newInstance() {
        return new HoldersFragment();
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater,
            @Nullable ViewGroup container,
            @Nullable Bundle savedInstanceState) {
        b = DataBindingUtil.inflate(inflater, R.layout.fragment_holders, container, false);
        return b.getRoot();
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        initUI();
    }

    @Override
    public void setHolders(List<Holder> holders) {
        viewModel.setHolders(holders);
    }

    @Override
    public void openHolderCards(Holder holder) {
        startActivity(HolderDetailsActivity.getIntent(getActivity(), holder));
    }

    @Override
    public void onItemClick(Holder item) {
        presenter.onHolderClick(item);
    }

    private void initUI() {
        viewModel = new HoldersViewModel();
        b.setModel(viewModel);
        b.setListener(this);
    }
}

File: presentation/src/main/kotlin/com/popalay/cardme/screens/holders/HoldersFragment.kt
package com.popalay.cardme.screens.holders

import android.arch.lifecycle.ViewModelProvider
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.popalay.cardme.R
import com.popalay.cardme.databinding.FragmentHoldersBinding
import com.popalay.cardme.base.BaseFragment
import com.popalay.cardme.utils.extensions.getDataBinding
import com.popalay.cardme.utils.extensions.getViewModel
import javax.inject.Inject

class HoldersFragment : BaseFragment() {

    companion object {
        fun newInstance() = HoldersFragment()
    }

    @Inject lateinit var factory: ViewModelProvider.Factory

    private lateinit var b: FragmentHoldersBinding

    override fun onCreateView(inflater: LayoutInflater,
                              container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        b = getDataBinding<FragmentHoldersBinding>(inflater, R.layout.fragment_holders, container)
        b.vm = getViewModel<HoldersViewModel>(factory)
        return b.root
    }
}
